src/components/test/Excerpt-test.js

```diff
@@ -137,23 +138,51 @@ describe('Excerpt', () => {
       assert.equal(getExcerptHeight(wrapper), 200);
     });
 
-    it("sets button's default state to un-expanded", () => {
-      const wrapper = createExcerpt({ inlineControls: true }, TALL_DIV);
-      const button = getToggleButton(wrapper);
-      assert.equal(button.prop('expanded'), false);
-      assert.equal(button.text(), 'More');
+    [undefined, 'Show more'].forEach(inlineControlExpandText => {
+      it("sets button's default state to un-expanded", () => {
+        const wrapper = createExcerpt(
+          { inlineControl: true, inlineControlExpandText },
+          TALL_DIV,
+        );
+        const button = getToggleButton(wrapper);
+        assert.equal(button.prop('expanded'), false);
+        assert.equal(button.text(), inlineControlExpandText ?? 'More');
+      });
     });
 
-    it("changes button's state to expanded when clicked", () => {
-      const wrapper = createExcerpt({ inlineControls: true }, TALL_DIV);
-      let button = getToggleButton(wrapper);
-      act(() => {
-        button.props().onClick();
+    [undefined, 'Show less'].forEach(inlineControlCollapseText => {
+      it("changes button's state to expanded when clicked", () => {
+        const wrapper = createExcerpt(
+          { inlineControl: true, inlineControlCollapseText },
+          TALL_DIV,
+        );
+        let button = getToggleButton(wrapper);
+        act(() => {
+          button.props().onClick();
+        });
+        wrapper.update();
+        button = getToggleButton(wrapper);
+        assert.equal(button.prop('expanded'), true);
+        assert.equal(button.text(), inlineControlCollapseText ?? 'Less');
+      });
+    });
+
+    [true, false].forEach(collapsed => {
+      it('allows to control Excerpt externally', () => {
+        const fakeOnToggleCollapsed = sinon.stub();
+        const wrapper = createExcerpt(
+          {
+            inlineControl: true,
+            collapsed,
+            onToggleCollapsed: fakeOnToggleCollapsed,
+          },
+          TALL_DIV,
+        );
+
+        getToggleButton(wrapper).props().onClick();
+
+        assert.calledWith(fakeOnToggleCollapsed, !collapsed);
```

Comment:
This test demonstrates an Excerpt witn `inlineControl: true` can still be controlled.

Reply:
This test verifies that `onToggleCollapsed` is called, but doesn't check that the excerpt respects the new value of `collapsed`. See my note about the logic in `Excerpt`.